app/controllers/admin/websites_controller.rb

```diff
@@ -228,16 +249,77 @@ def approve
+      if @website.deny!
         Event.log_event(Event.names[:website_denied], 'Website', @website.id, "Website #{@website.domain} denied at #{DateTime.now}", current_user.id)
         redirect_to admin_website_url(@website), notice: "Website #{@website.domain} was denied."
       else
         render :edit
       end
     end
 
+    def develop
+      if @website.start_development!
+        Event.log_event(Event.names[:website_start_development], 'Website', @website.id, "Website #{@website.domain} begins development at #{DateTime.now}", current_user.id)
+        redirect_to admin_website_url(@website), notice: "Website #{@website.domain} in now in development."
+      else
+        render :edit
+      end
+    end
+
+    def stage
+      if @website.stage!
+        Event.log_event(Event.names[:website_staged], 'Website', @website.id, "Website #{@website.domain} was updated to Staging at #{DateTime.now}", current_user.id)
+        redirect_to admin_website_url(@website), notice: "Website #{@website.domain} has been staged."
+      else
+        render :edit
+      end
+    end
+
+    def launch
+      if @website.launch!
+        Event.log_event(Event.names[:website_launched], 'Website', @website.id, "Website #{@website.domain} was launched at #{DateTime.now}", current_user.id)
+        redirect_to admin_website_url(@website), notice: "Website #{@website.domain} has been launched."
+      else
+        render :edit
+      end
+    end
+
+    def redirect
+      if @website.redirect!
+        Event.log_event(Event.names[:website_redirected], 'Website', @website.id, "Website #{@website.domain} was redirected at #{DateTime.now}", current_user.id)
+        redirect_to admin_website_url(@website), notice: "Website #{@website.domain} has been redirected."
+      else
+        render :edit
+      end
+    end
+
+    def archive
+      if @website.archive!
+        Event.log_event(Event.names[:website_archived], 'Website', @website.id, "Website #{@website.domain} was archived at #{DateTime.now}", current_user.id)
+        redirect_to admin_website_url(@website), notice: "Website #{@website.domain} has been archived."
+      else
+        render :edit
+      end
+    end
+
+    def decommission
+      if @website.decommission!
+        Event.log_event(Event.names[:website_decommissioned], 'Website', @website.id, "Website #{@website.domain} was decommissioned at #{DateTime.now}", current_user.id)
+        redirect_to admin_website_url(@website), notice: "Website #{@website.domain} has been decommissioned."
+      else
+        render :edit
+      end
+    end
+
+    def reset
+      if @website.reset!
+        Event.log_event(Event.names[:website_reset], 'Website', @website.id, "Website #{@website.domain} was reset at #{DateTime.now}", current_user.id)
+        redirect_to admin_website_url(@website), notice: "Website #{@website.domain} has been reset to newly_requested status."
+      else
+        render :edit
+      end
+    end
+
```

Comment:
Can we DRY this up a bit?

I am thinking we can define the valid transitions [:develop, :stage, :launch, etc] and define a single controller method to transition state.

Reply:
ya, refactoring this makes sense!